vmb/matrimony/admin.py

```diff
@@ -70,9 +72,107 @@ def queryset(self, request, queryset):
                 rounds_chanting__lt=int(8), rounds_chanting__gte=int(1)
             )
         if self.value() == "0":
+
             return queryset.filter(rounds_chanting__lte=int(0))
 
 
+class InputFilter(admin.SimpleListFilter):
+    template = "admin/input_filter.html"
+
+    def lookups(self, request, model_admin):
+        # Dummy, required to show the filter.
+        return ((),)
+
+    def choices(self, changelist):
+        # Grab only the "all" option.
+        all_choice = next(super().choices(changelist))
+        all_choice["query_parts"] = (
+            (k, v)
+            for k, v in changelist.get_filters_params().items()
+            if k != self.parameter_name
+        )
+        yield all_choice
+
+
+class AnnualIncomeRangeFilter(InputFilter):
+    title = "Annual Income"
+    request = None
+    parameter_name = "currency"
+    field_name = "annual_income"
+    template = "admin/input_filter.html"
+
+    def __init__(self, request, params, model, model_admin):
+        super().__init__(request, params, model, model_admin)
+
+        self.request = request
+
+        if self.parameter_name + "_from" in params:
+            value = (params.pop(self.parameter_name + "_from")).split()
+            from_income = value[0]
+            from_currency = value[1]
+            self.used_parameters[self.field_name + "_from"] = from_income
+            self.used_parameters[self.parameter_name + "_from"] = from_currency
+
+        if self.parameter_name + "_to" in params:
+            value = (params.pop(self.parameter_name + "_to")).split()
```

Comment:
You should not modify/pop `self.parameter_name + "_to"` or `_from`. That's why the applied filter values are not getting rendered in the `from` to `to` fields post filtering.

Reply:
This is the issue.

![image](https://user-images.githubusercontent.com/231684/80013643-da6dd200-84ec-11ea-80ea-60465cea4d58.png)

Reply:
Check this here:
https://github.com/jpsoffice/vmb/blob/master/vmb/matrimony/admin.py#L93

This value gets rendered in the filter form field.

But for querying, I am using `self.field_name + '_from'`, https://github.com/jpsoffice/vmb/blob/master/vmb/matrimony/admin.py#L107

Reply:
Ok. Thank you for the detailed review. Will make the necessary changes.